samples/Gallery/Pages/AutoCompleteBoxPage.fs

```diff
@@ -366,7 +358,8 @@ module AutoCompleteBoxPage =
                     VStack() {
                         TextBlock("MinimumPrefixLength: 1")
 
-                        AutoCompleteBox("", TextChanged, model.Capitals)
+                        //TODO the input text is not set to this. Do I misunderstand this parameter?
```

Comment:
I don't agree from a model-binding POV: If I have a model with a string property and want to bind it to the in- and output text `AutoCompleteBox`, I think I'd prefer a constructor param to preset the `AutoCompleteBox.TextProperty` from the model next to the `fn` that updates the model with the new text via `TextChangedEventArgs `. That would keep the binding in one place and the syntax similar to other controls.

If you do remove the param however, please make sure to expose the `AutoCompleteBox.TextProperty` via a new modifier extension so it can be bound to. Thank you!

Reply:
@h0lg
So the thing the text that is bounded to the `AutoCompleteBox` is the `AutoCompleteBox.SearchTextProperty`(But the setter is private so cannot use it) rather than the `AutoCompleteBox.TextProperty`.  Also adding the `AutoCompleteBox.TextProperty` to the widget constructor would be misleading as people will expect that to contain  the search text.

Note that `AutoCompleteBox` it is not meant  to be used like the `TexBox`

> If you do remove the param however, please make sure to expose the AutoCompleteBox.TextProperty via a new modifier extension so it can be bound to.

Yeah this is what we should do. So you can add an initial text to `AutoCompleteBox`

Reply:
@edgarfgp Not sure I'm picking up everything you're laying down. But I've added a Text property, a modifier for it and tested it in the examples. It presets the text like I want it to but seems to break the dropdown - the menu doesn't open any longer. Please have a look when you find the time - and many thanks!

Reply:
@h0lg I think I found way to make it work in all scenarios. Needs to test this more tomorrow(Too late for me today). Try my last commit.

Reply:
Look like `AutoCompleteBox` can be only initialized with the ItemsSource or Populator.

Reply:
@edgarfgp That works as intended on both the synchronous and async auto-completes - thanks a lot! I've cleaned up, added some doco and an open question concerning renaming `TextChanged` to just `Text`.

Reply:
@h0lg Yeah we can name just `Text` if you want, but we should keep the modifier `onTextChanged`.